### ISSUE TITLE
adding LipschitzNorm for Attention in GAT

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#76]
Can you implement for GAT this special Normalisation on the Attention operator

https://github.com/gdasoulas/LipschitzNorm/blob/master/normalizations/lipschitznorm.py

**Comment by thegodone** [comment]
Sorry to ask again but due to fact GAT is the best for Tox21 this normalization trick should even improve the performance can you test it ?

**Comment by thegodone** [comment]
I think this can should be useful : 
```
import tensorflow as tf
from tensorflow.keras.layers import Layer
from tensorflow.keras.backend import expand_dims, sqrt
from tensorflow.keras import backend as K

class LipschitzNorm(Layer):
    def __init__(self, att_norm=4, recenter=False, scale_individually=True, eps=1e-12):
        super(LipschitzNorm, self).__init__()
        self.att_norm = att_norm
        self.eps = eps
        self.recenter = recenter
        self.scale_individually = scale_individually

    def call(self, inputs):
        x, att, alpha, index = inputs
        att_l, att_r = att

        if self.recenter:
            mean = K.sum(x, axis=0) / tf.math.unsorted_segment_sum(data=tf.ones_like(x), segment_ids=index, num_segments=tf.reduce_max(index)+1)
            x = x - mean

        norm_x = K.sum(K.square(x), axis=-1)
        max_norm = sqrt(tf.math.segment_max(data=norm_x, segment_ids=index) + norm_x)

        if self.scale_individually == False:
            norm_att = self.att_norm * K.sqrt(K.sum(K.square(K.concatenate([att_l, att_r], axis=-1))))
        else:
            norm_att = self.att_norm * K.sqrt(K.sum(K.square(K.concatenate([att_l, att_r], axis=-1)), axis=-1))

        alpha = alpha / (norm_att * max_norm + self.eps)
        return alpha
```
